rp2040-hal/src/xosc.rs

```diff
@@ -89,13 +83,12 @@ impl CrystalOscillator<Disabled> {
     }
 
     /// Initializes the XOSC : frequency range is set, startup delay is calculated and set.
-    pub fn initialize(self, frequency: Hertz) -> Result<CrystalOscillator<Initialized>, Error> {
+    pub fn initialize(self, frequency: HertzU32) -> Result<CrystalOscillator<Initialized>, Error> {
         const ALLOWED_FREQUENCY_RANGE: RangeInclusive<Megahertz<u32>> =
-            Megahertz(1)..=Megahertz(15);
-        const STABLE_DELAY: Milliseconds = Milliseconds(1_u32);
-        const DIVIDER: Fraction = Fraction::new(256, 1);
+            MegahertzU32::MHz(1)..=MegahertzU32::MHz(15);
```

Comment:
If you use `const ALLOWED_FREQUENCY_RANGE: RangeInclusive<HertzU32> =
            HertzU32::MHz(1)..=HertzU32::MHz(15);`

you can lose the convert on line 91